### ISSUE TITLE
UTF-8 (and '...') handling

### ISSUE THREAD

**Issue by braoult** [antoinepirlot/Satunes#603]
After starting Satunes, having a look in `Albums` tab, the following get displayed :

![hor](https://github.com/user-attachments/assets/e9d90438-64f5-4562-a553-47d8ef7cf6c8)

The left faulty album name should show `Ainsi Soit Je...` (with 3 trailing dots, therefore only ASCII here).
The right faulty one should show `いぶき` (UTF-8).

**Comment by antoinepirlot** [comment]
Hey, 

Thanks for reporting.

Hmm, I thought it was already fixed 🤔.

I will check that, thanks

**Comment by braoult** [comment]
Thanks a lot. I forgot: I use `Android 14` .

**Comment by antoinepirlot** [comment]
> Thanks a lot. I forgot: I use `Android 14` .

Thanks

**Comment by braoult** [comment]
If you need, I can transfer you some faulty Music files, but this would not be possible here, due to © issues...

**Comment by antoinepirlot** [comment]
> If you need, I can transfer you some faulty Music files, but this would not be possible here, due to © issues...

You can send me by email if you prefer at pirlot.antoine@outlook.com

**Comment by antoinepirlot** [comment]
Issue duplicated for ellipsis except for いぶき.

I also noticed it happens not everytime. I set the album name: "Ainsi Soit Je..." with the elipsis char will show wrong chars instead of "..."
but "いぶき ..." (still with ellipsis char) won't show wrong chars.

It's due to formatted chars in files informations by the OS with the one you edit names.

Also, I'll check to make app accepting different formats to avoid this issue, later.

**Comment by antoinepirlot** [comment]
Not enough knowledge at this time

**Comment by antoinepirlot** [comment]
I don't have enough knowledge about that at this time

**Comment by braoult** [comment]
Usually, to fix UTF-related issues, one can first dump the data and understand what characters are exactly.

For the "Ainsi Soit Je…" case, I just double-checked, and dumped the data (filenames and Mp3 tags) from the sample I shared with you.

My mistake, some of the ellipsis for this case are not ASCII, but the `HORIZONTAL ELLIPSIS` UTF-8 character ([U+2026](https://unicode-explorer.com/c/2026)) :
- The ellipsis in directory name "`Mylène Farmer/Ainsi Soit Je…/`" uses U+2026
- In "*L'Horloge*" MP3, the `TALB` (album) tag ("*Ainsi Soit Je…*"), the ellipsis is also the U+2026 character.

This *Album* MP3 tag should be decoded (maybe unnecessary) and displayed as standard UTF-8. How do you proceed in this case to get a wrong display ?

Note: U+2026 is hex: `e2 80 a6`

**Comment by antoinepirlot** [comment]
Yeah, I spent a lot of times to get a solution that doesn't make Satunes loads longer.

I checked about id3v2 but I didn't find a way to manage the encoding type with no huge performance impact.

Also, I'm begging a master degree in computer science, I hope I will find a solution with more knowledge 🤭.

I'm still checking for a solution.
